feat: Show the current users name in the sidebar

JonathanBout opened the pull request:
**Description**
This pull request adds a line to the sidebar with the username of the currently logged in user:
<img width="360" height="540" alt="sidebar with admin username" src="https://github.com/user-attachments/assets/519b6144-a032-4833-aa60-d053e16a9566" />
When clicked, you will be sent to the user settings.

This PR also modifies the behavior of the settings button, which now redirects to the global settings and will be hidden when the current user doesn't have the rights for that page.

To make this possible, the backend is modified to also send the username which wasn't sent before.

(resolves https://github.com/filebrowser/filebrowser/issues/2692)

JonathanBout committed:
Add username in Sidebar

JonathanBout committed:
Merge branch 'master' of https://github.com/filebrowser/filebrowser

JonathanBout committed:
Update README.md

JonathanBout committed:
Merge branch 'master' into add-username-in-sidebar

JonathanBout committed:
Merge branch 'filebrowser:master' into master

JonathanBout committed:
Merge branch 'master' into add-username-in-sidebar

JonathanBout committed:
Merge branch 'master' into add-username-in-sidebar

JonathanBout committed:
Merge branch 'master' into add-username-in-sidebar

JonathanBout committed:
Merge branch 'filebrowser:master' into add-username-in-sidebar

JonathanBout committed:
Merge branch 'master' into add-username-in-sidebar

JonathanBout committed:
Merge branch 'master' into add-username-in-sidebar

o1egl committed:
Merge branch 'master' into add-username-in-sidebar

o1egl commented on frontend/src/components/Sidebar.vue:
```diff
@@ -33,7 +48,7 @@
       </div>
 
-      <div>
+      <div v-if="user.perm.admin">
```
Why if that added?

JonathanBout replied:
Because the `toSettings` function now navigates to `settings/global`, which is only accessible to an admin right?

o1egl commented on frontend/src/components/Sidebar.vue:
```diff
@@ -1,5 +1,20 @@
+    >
+      <i class="material-icons">exit_to_app</i>
+      <span>{{ $t("sidebar.logout") }}</span>
+    </button>
```
Why is it moved?

JonathanBout replied:
To me it made more sense to have profile related buttons together

o1egl replied:
Logout is not a common thing that users want to do, so it should be placed on top as a main action

JonathanBout replied:
You mean should _not_ right?

Is it better if it's moved to the bottom of the sidebar?

o1egl reviewed (commented):
I don't think addind user and especially logout to the top of the side panel is a good ux practice. There should be a better way to visualize it

JonathanBout committed:
Merge branch 'master' of https://github.com/JonathanBout/filebrowser into add-username-in-sidebar

JonathanBout committed:
move logout button to the bottom of the sidebar

JonathanBout committed:
Remove empty arrow functions that aren't needed

JonathanBout committed:
Merge branch 'master' into add-username-in-sidebar

JonathanBout committed:
Merge branch 'master' into add-username-in-sidebar

JonathanBout committed:
Merge branch 'master' into add-username-in-sidebar

JonathanBout committed:
Merge branch 'master' into add-username-in-sidebar

JonathanBout committed:
Merge branch 'master' into add-username-in-sidebar

JonathanBout committed:
Merge branch 'master' into add-username-in-sidebar

JonathanBout committed:
Merge branch 'master' into add-username-in-sidebar

JonathanBout committed:
Merge branch 'master' into add-username-in-sidebar

JonathanBout committed:
Merge branch 'master' into add-username-in-sidebar

JonathanBout committed:
Merge branch 'master' into add-username-in-sidebar

JonathanBout committed:
Merge branch 'master' into add-username-in-sidebar

hacdias committed:
Merge branch 'master' into add-username-in-sidebar

hacdias commented on frontend/src/components/Sidebar.vue:
```diff
@@ -1,6 +1,10 @@
 <template>
   <div v-show="active" @click="closeHovers" class="overlay"></div>
   <nav :class="{ active }">
+    <button v-if="user.username" @click="toAccountSettings" class="action">
```
Feeling a bit iffy about this `v-if`. Why would `user.username` not be set?

JonathanBout replied:
First of all, thank you for the review!

As it's been more than a year since I wrote this code, I don't really know anymore 😅 but it might have been something to do with guest mode, where the user isn't logged in and thus doesn't have a username

JonathanBout replied:
After some testing I found it indeed is not needed, after moving the button into the `<template v-if="isLoggedIn">`.
This happened due to changes to the file after I wrote the code, in my initial commit ([`1909312`](https://github.com/filebrowser/filebrowser/pull/2821/commits/1909312b8958dd37cc1440219243062d7369d518)) it still made sense.

JonathanBout committed:
Merge branch 'master' into add-username-in-sidebar

JonathanBout committed:
cleanup & resolve comment

- remove username check, move current user indicator into v-if=loggedIn
- use the closeHovers method instead of committing to the store

JonathanBout committed:
Merge branch 'master' into add-username-in-sidebar

hacdias committed:
refactor: rename toSettings

hacdias committed:
Merge branch 'master' into add-username-in-sidebar

hacdias reviewed (approved):
I am approving this because I think it is a good idea and that it is useful to show the user's username somewhere in the sidebar. However, I feel like it will cause some confusion for existing users regarding the Settings button being missing.

If that happens to be the case, we can reinstate the settings button for all users. However, I think making it clear in the changelog should be enough.

<img width="622" height="582" alt="Image" src="https://github.com/user-attachments/assets/8bf8104b-396e-4955-a63b-163177985868" />
